sdks/python/apache_beam/examples/ml_transform/ml_transform_basic.py

```diff
@@ -61,49 +61,89 @@ def parse_args():
   return parser.parse_known_args()
 
 
-def run(args):
-  data = [
-      dict(x=["Let's", "go", "to", "the", "park"]),
-      dict(x=["I", "enjoy", "going", "to", "the", "park"]),
-      dict(x=["I", "enjoy", "reading", "books"]),
-      dict(x=["Beam", "can", "be", "fun"]),
-      dict(x=["The", "weather", "is", "really", "nice", "today"]),
-      dict(x=["I", "love", "to", "go", "to", "the", "park"]),
-      dict(x=["I", "love", "to", "read", "books"]),
-      dict(x=["I", "love", "to", "program"]),
-  ]
-
+def preprocess_data_for_ml_training(train_data, artifact_mode, args):
   with beam.Pipeline() as p:
```

Comment:
We need some notes that this is the first of two pipelines that a user will run .

Reply:
I think I agree that having a function header that describes it as a complete pipeline run would be helpful for folks who are less familiar with Beam (and a similar comment on the second one). Running 2 pipelines in a single example might be unexpected